packages/flutter/lib/src/painting/_network_image_io.dart

```diff
@@ -122,3 +122,25 @@ class NetworkImage extends image_provider.ImageProvider<image_provider.NetworkIm
   @override
   String toString() => '$runtimeType("$url", scale: $scale)';
 }
+
+/// Thrown to indicate that loading a network image failed
+///
+class NetworkImageLoadException implements Exception {
+  /// Creates a [NetworkImageLoadException] with the specified http status
+  /// [code] and the [uri]
```

Comment:
nit: end with `.`.

We should document here that `uri` must not be null.  Should also either remove `@required` from `code`, or `assert(code != null)`.

Reply:
I would go with `assert(code != null)`

Reply:
This one may still need attention - and now `[code]` should be updated to `[statusCode]`.

Thanks for bearing with us on this!